### PR TITLE
[ACS-5748] The file gets unshared when setting the past expiration date

### PR DESCRIPTION
**Pull request by nikita-web-ua**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

- Currently it is possible to type in past date for link expiration, which leads to an request error and file is being unshared
- Input is not filtering alphabets.
- Currently API request being triggered on every change to the input, so it is impossible to set fully the date manually as you constantly being unfocused from the input.

**What is the new behaviour?**

- Added input validation to prevent old date from being submitted.
- Now the user cannot enter alphabets in the date picker input.
- Now form is being submitted by 'enter' or blur event, which allows to type in the date fully 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://alfresco.atlassian.net/browse/ACS-5748

### CODE REVIEW

**Review by DenysVuika — APPROVED** [review]

**Comment by DenysVuika** [comment]
needs rebase

File: lib/content-services/src/lib/content-node-share/content-node-share.dialog.spec.ts [hunk 1]
```diff
@@ -45,6 +45,17 @@ describe('ShareDialogComponent', () => {
...
 
+    const fillInDatepickerInput = (value: string) => {
+        const input = fixture.debugElement.query(By.css('.adf-share-link__input')).nativeElement;
+        input.value = value;
+        input.dispatchEvent(new Event('input'));
+        input.dispatchEvent(new Event('blur'));
+        fixture.detectChanges();
+    };
+
+    const clickExpireToggleButton = () => fixture.nativeElement.querySelector('mat-slide-toggle[data-automation-id="adf-expire-toggle"] label')
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/content-node-share/content-node-share.dialog.spec.ts:56]: Could you reuse that in 333rd line as there is the same logic?

File: lib/content-services/src/lib/content-node-share/content-node-share.dialog.spec.ts [hunk 2]
```diff
@@ -342,5 +347,44 @@ describe('ShareDialogComponent', () => {
...
+            fixture.detectChanges();
+            clickExpireToggleButton();
+            fillInDatepickerInput('32');
+            const error = fixture.debugElement.query(By.css('[data-automation-id="adf-share-link-input-warning"]'));
+
+            expect(error).toBeTruthy();
+            expect(component.time.hasError('invalidDate')).toBeTrue();
+        });
+
+        it('should show an error when provided date is valid', () => {
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/content-node-share/content-node-share.dialog.spec.ts:380]: incorrect title, should be: "should not show ..."

File: lib/content-services/src/lib/content-node-share/content-node-share.dialog.spec.ts (hunk 1 above)
Comment (nikita-web-ua) [reply lib/content-services/src/lib/content-node-share/content-node-share.dialog.spec.ts:56]: adressed, thanks

File: lib/content-services/src/lib/content-node-share/content-node-share.dialog.spec.ts (hunk 2 above)
Comment (nikita-web-ua) [reply lib/content-services/src/lib/content-node-share/content-node-share.dialog.spec.ts]: corrected, ty

**Pull request merged**